### ISSUE TITLE
Codemirror shouldn't error if an elided binding is actually set in the front matter

### ISSUE THREAD

**Issue by hardyjosh** [rainlanguage/rain.orderbook#587]
![telegram-cloud-photo-size-4-5818672241100046206-y](https://github.com/rainlanguage/rain.orderbook/assets/1190022/5b5995d3-3d61-49af-a173-d9ea2697b4d3)

this shouldn't error if the binding is set in the frontmatter

**Comment by rouzwelt** [comment]
- dotrain doesnt know about any of the data in frontmatter
- any bindings (or in general any data) from frontmatter should be explicitly given to the dotrain logic 
- codemirror errors are provided as a callback, so the underlying RainDocument with any bindings are processed in rust and then array of Problems are provided as callback result to the codemirror, this means that any filtration on Problems or any bindings for the underlying RainDcouemnt should be handled in `ob lsp order` realted logic, I think it is in common crate in ob repo
- `DotrainAddOrderLsp` struct already receives any optional bindings, so I think for fixing this, the provided bindings just need to be fixed to contain any frontmatter bindings

**Comment by hardyjosh** [comment]
Agreed on the solution, but I think we'd need to scoop up any and all bindings from all of the scenarios to pass to codemirror. So the union of all possible bindings set in the front matter.

**Comment by rouzwelt** [comment]
based on discussion on tg, better to off codemirror errors for all elided bindings, that's an easy fix, just need to filter out those that have `elided binding's error code`

**Comment by hardyjosh** [comment]
Yes we can do this for now to squash these errors as they aren't that useful and are really noisy.

In future we need to tackle this more broadly - the fact that we have multiple scenarios represented in the same code.

**Comment by hardyjosh** [comment]
@rouzwelt pointed out that this will interfere with the parsing pipeline for the app - dotrain compse > native parser

so we need to revisit this with a better plan for gui parsing in general
